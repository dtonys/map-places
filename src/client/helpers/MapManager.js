import lodashGet from 'lodash/get';
import {
  appendScriptToHead,
} from 'helpers/domUtils';
import uuidv1 from 'uuid/v1';
import lodashFindIndex from 'lodash/findIndex';
import lodashNoOp from 'lodash/noop';


export const BASIC_MAP_TYPE_ROADMAP = 'roadmap';
export const BASIC_MAP_TYPE_SATELLITE = 'satellite';
export const BASIC_MAP_TYPE_HYBRID = 'hybrid';
export const BASIC_MAP_TYPE_TERRAIN = 'terrain';
export const BASIC_MAP_TYPES = [
  BASIC_MAP_TYPE_ROADMAP,
  BASIC_MAP_TYPE_SATELLITE,
  BASIC_MAP_TYPE_HYBRID,
  BASIC_MAP_TYPE_TERRAIN,
];

export function getIsGoogleMapsLoaded() {
  return (
    typeof window !== 'undefined' &&
    Boolean( lodashGet( window, 'google.maps.Map' ) )
  );
}

/*
** Control for displaying google maps UI
** Controls guide:
** https://developers.google.com/maps/documentation/javascript/controls
**
** `ControlPosition` constants:
** https://developers.google.com/maps/documentation/javascript/reference#ControlPosition
*/
const getMapControlOptionsDefault = () => {
  return {
    // mapTypeControl: Map / Satellite toggle
    mapTypeControl: false,
    mapTypeControlOptions: {
      style: google.maps.MapTypeControlStyle.HORIZONTAL_BAR,
      position: google.maps.ControlPosition.TOP_CENTER,
    },
    // zoomControl: + / - toggle
    zoomControl: true,
    zoomControlOptions: {
      position: google.maps.ControlPosition.RIGHT_BOTTOM,
    },
    // scaleControl: 2KM distance in legend
    scaleControl: false,
    // streetViewControl: enables pegman
    streetViewControl: false,
    streetViewControlOptions: {
      position: google.maps.ControlPosition.LEFT_TOP,
    },
    // fullscreenControl: allow full screen
    fullscreenControl: false,
  };
};

const ICON_PATH = '/static/pin_w30.svg';

const MapManager = {
  /** Private members **/
  _mapDomNode: null,

  /** Public members **/
  initializeComplete: null,
  googleMap: null,
  googleMapMarkers: {},
  onMapClickListeners: [],
  onMarkerClickListeners: [],

  /** Private methods **/
  _createMapDomNode: function () { // eslint-disable-line func-names
    if ( this._mapDomNode ) return;
    this._mapDomNode = document.createElement('div');
    this._mapDomNode.style.cssText = 'height: 100%; width: 100%;';
  },
  _loadGoogleMapsScript: function () { // eslint-disable-line func-names
    return new Promise((resolve) => {
      const isGoogleMapsScriptLoaded = getIsGoogleMapsLoaded();
      if ( isGoogleMapsScriptLoaded ) {
        resolve();
        return;
      }
      window.initMap = resolve;
      appendScriptToHead(`https://maps.googleapis.com/maps/api/js?v=3&key=${GOOGLE_MAPS_API_KEY}&callback=initMap&libraries=places`);
    });
  },
  _createGoogleMap: function ({ // eslint-disable-line func-names
    lat,
    lng,
    mapTypeId = BASIC_MAP_TYPE_ROADMAP,
  }) { // eslint-disable-line func-names
    if ( this.googleMap ) return;
    this.googleMap = new google.maps.Map(this._mapDomNode, { // eslint-disable-line
      center: { lat, lng },
      zoom: 10,
      mapTypeId,
      ...getMapControlOptionsDefault(),
    });
    this.googleMap.addListener('click', (event) => {
      const clickedLat = event.latLng.lat();
      const clickedLng = event.latLng.lng();

      console.log(`clicked map at ${clickedLat} ${clickedLng} `); // eslint-disable-line no-console
      this.onMapClickListeners.forEach((fnObj) => {
        fnObj.onClickFn({
          lat: clickedLat,
          lng: clickedLng,
        });
      });
    });

    // NOTE: Expose for debugging purposes.
    window.googleMap = this.googleMap;
  },

  /** Public methods **/
  initialize: function ({ initialCenterLatLng }) { // eslint-disable-line func-names
    this._createMapDomNode();
    this.initializeComplete = this._loadGoogleMapsScript()
      .then(() => {
        this._createGoogleMap( initialCenterLatLng );
      });
  },

  insertMapToDom: function insertMapToDom( containerDomNode ) {
    containerDomNode.appendChild(this._mapDomNode);
    // NOTE: Trigger a resize to make the map appear
    google.maps.event.trigger(this.googleMap, 'resize'); // eslint-disable-line
    return true;
  },
  attachMarkerToMap: function attachMarkerToMap({ lat, lng }) {
    const latLngUID = `${lat}_${lng}`;
    if ( this.googleMapMarkers[latLngUID] ) {
      return lodashNoOp;
    }
    const marker = new google.maps.Marker({
      position: { lat, lng },
      map: this.googleMap,
      icon: ICON_PATH,
    });
    marker.addListener('click', () => {
      const latLngFns = marker.getPosition();
      const clickedLat = latLngFns.lat();
      const clickedLng = latLngFns.lng();
      console.log(`clicked marker at ${clickedLat}, ${clickedLng} `); // eslint-disable-line no-console
      this.onMarkerClickListeners.forEach((fnObj) => {
        fnObj.onClickFn({
          marker,
          latLngUID,
        });
      });
    });
    this.googleMapMarkers[latLngUID] = marker;
    return {
      marker,
      latLngUID,
    };
  },
  removeMarkerFromMap: function (latLngUID) { // eslint-disable-line func-names
    const marker = this.googleMapMarkers[latLngUID];
    if ( !marker ) {
      return;
    }
    this.googleMapMarkers[latLngUID].setMap(null);
    delete this.googleMapMarkers[latLngUID];
  },
  addOnMapClickListener: function ( onClickFn ) { // eslint-disable-line func-names
    const fnId = uuidv1();
    this.onMapClickListeners.push({
      id: fnId,
      onClickFn,
    });
    // return onRemove
    return function removeMapClickListener() {
      const index = lodashFindIndex( this.onMapClickListeners, { id: fnId });
      if ( index !== -1 ) {
        const removedFn = this.onMapClickListeners.splice(index, 1);
        return removedFn;
      }
      return lodashNoOp;
    };
  },
  addOnMarkerClickListener: function (onClickFn) { // eslint-disable-line func-names
    const fnId = uuidv1();
    this.onMarkerClickListeners.push({
      id: fnId,
      onClickFn,
    });
    // return onRemove
    return function removeMarkerClickListener() {
      const index = lodashFindIndex( this.onMarkerClickListeners, { id: fnId });
      if ( index !== -1 ) {
        const removedFn = this.onMarkerClickListeners.splice(index, 1);
        return removedFn;
      }
      return lodashNoOp;
    };
  },

  resetMap: function () { // eslint-disable-line func-names
    this.deleteAllMarkers();
    this.removeAllListeners();
  },

  deleteAllMarkers: function () { // eslint-disable-line func-names
    this.googleMapMarkers.forEach((marker) => {
      marker.setMap(null);
    });
    this.googleMapMarkers = [];
  },

  removeAllListeners: function () { // eslint-disable-line func-names
    this.onMapClickListeners = [];
    this.onMarkerClickListeners = [];
  },
};

export default MapManager;
