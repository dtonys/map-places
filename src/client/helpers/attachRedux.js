import { Component } from 'react';
import PropTypes from 'prop-types';
import { wrapDisplayName } from 'recompose';
import { Provider as ReduxStoreProvider } from 'react-redux';

import createStore from 'redux-modules/createStore';
import makeAction, {
  execute,
} from 'helpers/reduxAction';

import {
  ACTION_INCREMENT_COUNTER,
} from 'redux-modules/actions/counter';

let clientStore = null;
let serverStore = null;
let serverStoreInitialState = null;
function AttachReduxWithArgs(/* args */) {

  function AttachRedux( WrappedComponent ) {

    class AttachReduxHOC extends Component {
      static propTypes = {
        serverStoreInitialState: PropTypes.object,
      }

      static async getInitialProps( nextJSContext ) {

        if ( __SERVER__ ) {
          const { res } = nextJSContext;
          serverStore = createStore();
          serverStore.dispatch( makeAction(
            execute(ACTION_INCREMENT_COUNTER)
          ) );
          serverStoreInitialState = serverStore.getState();
          // Pass the store to the child components, for SSR.
          res.locals.reduxStore = serverStore;
        }
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(nextJSContext);
        }
        return {
          ...wrappedComponentInitialProps,
          serverStoreInitialState,
        };
      }

      constructor( props ) {
        super(props);
        if ( __CLIENT__ ) {
          if ( !clientStore ) {
            clientStore = createStore( props.serverStoreInitialState );
            window.store = clientStore;
          }
        }
      }

      render() {
        const store = serverStore || clientStore;

        return (
          <ReduxStoreProvider store={store} >
            <WrappedComponent {...this.props} />
          </ReduxStoreProvider>
        );
      }
    }

    AttachReduxHOC.displayName = wrapDisplayName( WrappedComponent, 'AttachReduxHOC' );
    return AttachReduxHOC;
  }
  return AttachRedux;
}
export default AttachReduxWithArgs;
