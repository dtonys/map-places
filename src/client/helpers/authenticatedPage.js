import { Component } from 'react';
import { wrapDisplayName } from 'recompose';
import lodashDifference from 'lodash/difference';

import { Router } from 'routes/pageRoutes';

import {
  extractCurrentUser,
} from 'redux-modules/reducers/user';

// @authenticated()
// @authenticated({ requiredRoles: ['admin', 'member'] })
function AuthenticatedPageWithOptions( options = {} ) {

  function AuthenticatedPage( WrappedComponent ) {

    class AuthenticatedPageHOC extends Component {
      static authRedirect( context, authenticated ) {
        if ( __SERVER__ ) {
          const { req, res } = context;
          const redirectPath = authenticated ? '/' : `/login?next=${encodeURIComponent(req.originalUrl)}`;
          res.redirect(redirectPath);
        }
        if ( __CLIENT__ ) {
          const redirectPath = authenticated ? '/' : `/login?next=${encodeURIComponent(context.asPath)}`;
          Router.replaceRoute(redirectPath);
        }
      }

      static async getInitialProps( context ) {
        const { store } = context;
        const globalState = store.getState();

        const currentUser = extractCurrentUser(globalState);
        if ( !currentUser ) {
          AuthenticatedPageHOC.authRedirect(context, false);
          return {};
        }
        if (
          options.requiredRoles &&
          lodashDifference(options.requiredRoles, currentUser.roles).length !== 0
        ) {
          AuthenticatedPageHOC.authRedirect(context, Boolean(currentUser));
          return {};
        }
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(context);
        }
        return wrappedComponentInitialProps;
      }

      render() {
        return <WrappedComponent {...this.props} />;
      }

    }
    AuthenticatedPageHOC.displayName = wrapDisplayName( WrappedComponent, 'AuthenticatedPageHOC' );
    return AuthenticatedPageHOC;
  }
  return AuthenticatedPage;
}

export default AuthenticatedPageWithOptions;
