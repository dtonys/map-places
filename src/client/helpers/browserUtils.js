export function tryGetCurrentPosition() {
  return new Promise((resolve, reject) => {
    // HTML5 geolocation
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition((position) => {
        resolve({
          lat: position.coords.latitude,
          lng: position.coords.longitude,
        });
      }, () => {
        console.log('geolocation.getCurrentPosition failed.'); // eslint-disable-line no-console
        reject();
      });
    }
    else {
      console.log('Browser doesn\'t support Geolocation.'); // eslint-disable-line no-console
      reject();
    }
  });
}

export function placeHolder() { }
