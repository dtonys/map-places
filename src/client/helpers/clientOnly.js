import { Component } from 'react';
import { wrapDisplayName } from 'recompose';


function clientOnly( WrappedComponent ) {

  class clientOnlyHOC extends Component {

    constructor( props ) {
      super(props);
      this.state = {
        clientLoaded: false,
      };
    }

    componentDidMount() {
      this.setState({ // eslint-disable-line
        clientLoaded: true,
      });
    }

    render() {
      const { clientLoaded } = this.state;
      if ( clientLoaded ) {
        return ( <WrappedComponent {...this.props} /> );
      }
      return (<div></div>);
    }
  }
  clientOnlyHOC.displayName = wrapDisplayName( WrappedComponent, 'clientOnlyHOC' );
  return clientOnlyHOC;
}

export default clientOnly;
