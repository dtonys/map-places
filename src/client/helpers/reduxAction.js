export const REQUESTED_SUFFIX = '_REQUESTED';
export const STARTED_SUFFIX = '_STARTED';
export const SUCCESS_SUFFIX = '_SUCCESS';
export const ERROR_SUFFIX = '_ERROR';
export const EXECUTED_SUFFIX = '_EXECUTED';

export default function makeAction( type, payload, meta = {} ) {
  return {
    type,
    payload,
    meta,
  };
}

// Send a request to the saga.  Only one may run at a time.
export function request( type ) {
  return `${type}${REQUESTED_SUFFIX}/request`;
}

// Send a request to the saga.  Many can run in parallel.
export function requestFork( type ) {
  return `${type}${REQUESTED_SUFFIX}/requestFork`;
}

// Report an API start
export function apiStart( type ) {
  return `${type}${STARTED_SUFFIX}/apiStart`;
}

// Report an API success
export function apiSuccess( type ) {
  return `${type}${SUCCESS_SUFFIX}/apiSuccess`;
}

// Report an API error
export function apiError( type ) {
  return `${type}${STARTED_SUFFIX}/apiError`;
}

// Report a timeout start
export function timeoutStart( type ) {
  return `${type}${STARTED_SUFFIX}/timeoutStart`;
}

// Report a timeout end
export function timeoutEnd( type ) {
  return `${type}${STARTED_SUFFIX}/timeoutEnd`;
}

// Execute a state change, goes directly to the reducer with no side effects
export function execute( type ) {
  return `${type}${EXECUTED_SUFFIX}/execute`;
}
