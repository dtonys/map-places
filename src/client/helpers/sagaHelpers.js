import {
  put,
  call,
  take,
  fork,
} from 'redux-saga/effects';

import makeAction, {
  request,
  requestFork,
  apiStart,
  apiSuccess,
  apiError,
  localStorageStart,
  localStorageSuccess,
  localStorageError,
} from 'helpers/reduxAction';


// export function* apiFlow( api, actionName, deferred, isLocalStorage ) {
export function* apiFlow( api, actionName, { deferred, isLocalStorage }) {
  const start = isLocalStorage ? localStorageStart : apiStart;
  const success = isLocalStorage ? localStorageSuccess : apiSuccess;
  const error = isLocalStorage ? localStorageError : apiError;

  let responseData = null;
  const dispatchedActionsTracker = {
    log: [],
    startAction: null,
    successAction: null,
    errorAction: null,
    addAction: function ( trackerType, action ) { // eslint-disable-line func-names
      if ( ~Object.keys(this).indexOf(trackerType) ) {
        this[trackerType] = action;
      }
      this.log.push( action );
    },
  };

  const _startedAction = makeAction( start(actionName) );
  yield put( _startedAction );
  dispatchedActionsTracker.addAction('startAction', _startedAction);
  try {
    responseData = yield call( api );
    const _successAction = makeAction( success(actionName), responseData );
    yield put( _successAction );
    dispatchedActionsTracker.addAction('successAction', _successAction);
    if ( deferred ) {
      deferred.resolve(_successAction);
    }
  }
  catch ( errorData ) {
    const _errorAction = makeAction( error(actionName), errorData );
    yield put( _errorAction );
    responseData = errorData;
    dispatchedActionsTracker.addAction('errorAction', _errorAction);
    if ( deferred ) {
      deferred.reject(_errorAction);
    }
  }

  return dispatchedActionsTracker;
}

export function createSagaWatcher( actionType, actionRunner ) {
  return function* () { // eslint-disable-line func-names
    while ( true ) { // eslint-disable-line no-constant-condition
      const action =  yield take([
        actionType,
        request(actionType),
        requestFork(actionType),
      ]);
      if ( action.type === request(actionType) ) {
        yield* actionRunner( action );
      }
      if ( action.type === requestFork(actionType) ) {
        fork( actionRunner, action );
      }
    }
  };
}

