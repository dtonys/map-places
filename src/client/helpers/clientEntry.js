// HOC for client side entry
import { Component } from 'react';
import { wrapDisplayName } from 'recompose';

import MapManager from 'helpers/MapManager';


let clientEntryRun = false;
// NOTE: Perform client side initialization here.  This is run once only.
function clientEntry() {
  MapManager.initialize({
    initialCenterLatLng: {
      lat: 37.549854,
      lng: -122.291353,
    },
  });
}


function ClientEntryWithArgs(/* args */) {

  function ClientEntry( WrappedComponent ) {

    class ClientEntryHOC extends Component {
      static async getInitialProps( nextJSContext ) {
        let wrappedComponentInitialProps = {};
        if ( WrappedComponent.getInitialProps ) {
          wrappedComponentInitialProps = await WrappedComponent.getInitialProps(nextJSContext);
        }
        return wrappedComponentInitialProps;
      }

      constructor(props) {
        super(props);
        // Run client entry, once
        if ( __CLIENT__ ) {
          if ( !clientEntryRun ) {
            clientEntry();
            clientEntryRun = true;
          }
        }
      }

      render() {
        return <WrappedComponent {...this.props} />;
      }
    }

    ClientEntryHOC.displayName = wrapDisplayName( WrappedComponent, 'ClientEntryHOC' );
    return ClientEntryHOC;
  }
  return ClientEntry;
}

export default ClientEntryWithArgs;
