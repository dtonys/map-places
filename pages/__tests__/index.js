import { getMockUrlProp } from './helpers/utils';
import { mount } from 'enzyme';
import HomePage from '../index';


describe('Home page', () => {

  let homePage = null;
  beforeAll((done) => {
    homePage = mount(<HomePage url={getMockUrlProp('/')} />);
    done();
  });

  test('Should mount and render successfully', (done) => {
    expect(homePage).toMatchSnapshot();
    done();
  });
});
