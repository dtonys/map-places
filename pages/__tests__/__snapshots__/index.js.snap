// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Should mount and render successfully 1`] = `
<ClientEntryHOC(AttachReduxHOC(Connect(HomePage)))
  url={
    Object {
      "asPath": "/",
      "pathname": "/",
      "query": Object {},
    }
  }
>
  <AttachReduxHOC(Connect(HomePage))
    url={
      Object {
        "asPath": "/",
        "pathname": "/",
        "query": Object {},
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(HomePage)
        url={
          Object {
            "asPath": "/",
            "pathname": "/",
            "query": Object {},
          }
        }
      >
        <HomePage
          dispatch={[Function]}
          places={Object {}}
          placesLoadedFromStorage={false}
          url={
            Object {
              "asPath": "/",
              "pathname": "/",
              "query": Object {},
            }
          }
        >
          <div
            className="jsx-826859815"
          >
            <Layout
              htmlTitle="Home"
            >
              <div
                className="jsx-2650881538"
              >
                <SideEffect(Head)>
                  <Head />
                </SideEffect(Head)>
                <Connect(Navbar)>
                  <Navbar
                    currentUser={null}
                    dispatch={[Function]}
                    userAuthenticated={false}
                  >
                    <div
                      className="jsx-306046629 navbar"
                    >
                      <div
                        className="jsx-306046629 section section-1"
                      >
                        <LinkRoutes
                          route="/"
                        >
                          <Link
                            as="/"
                            href="/"
                          >
                            <a
                              className="jsx-306046629 logoLink"
                              href="/"
                              onClick={[Function]}
                            >
                              <div
                                className="jsx-306046629 logoName"
                              >
                                Map Places
                              </div>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <div
                          className="jsx-306046629 item"
                        >
                          Maps
                        </div>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <div
                          className="jsx-306046629 item"
                        >
                          Other
                        </div>
                      </div>
                      <div
                        className="jsx-306046629 section section-2"
                      />
                      <div
                        className="jsx-306046629 section section-3"
                      >
                        <LinkRoutes
                          route="/login"
                        >
                          <Link
                            as="/login"
                            href="/login"
                          >
                            <a
                              className="jsx-306046629"
                              href="/login"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline"
                              >
                                Login
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <div
                          className="jsx-306046629"
                          style={
                            Object {
                              "marginRight": "10px",
                            }
                          }
                        />
                        <LinkRoutes
                          route="/signup"
                        >
                          <Link
                            as="/signup"
                            href="/signup"
                          >
                            <a
                              className="jsx-306046629"
                              href="/signup"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline secondary"
                              >
                                Signup
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                        <LinkRoutes
                          route="/users/profile"
                        >
                          <Link
                            as="/users/profile"
                            href="/users?id=profile"
                          >
                            <a
                              className="jsx-306046629"
                              href="/users/profile"
                              onClick={[Function]}
                            >
                              <button
                                className="jsx-306046629 button outline secondary"
                              >
                                Profile
                              </button>
                            </a>
                          </Link>
                        </LinkRoutes>
                      </div>
                      <JSXStyle
                        css=".navbar.jsx-306046629{height:60px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0 20px;background:#eee;position:absolute;top:0;left:0;right:0;border-bottom:solid black 2px;}.logoName.jsx-306046629{font-size:20px;color:rgb(49,52,57);cursor:pointer;text-decoration:none;}.logoLink.jsx-306046629{text-decoration:none;}.section-1.jsx-306046629{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-right:20px;white-space:nowrap;}.section-2.jsx-306046629{-webkit-flex:1;-ms-flex:1;flex:1;}.section-3.jsx-306046629{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;margin-left:20px;}.userName.jsx-306046629{margin-top:10px;overflow:hidden;text-overflow:ellipsis;max-width:200px;white-space:nowrap;}.searchWrap.jsx-306046629{margin:0 20px;}"
                        styleId="306046629"
                      />
                    </div>
                  </Navbar>
                </Connect(Navbar)>
                <div
                  className="jsx-2650881538 scrollWrap"
                >
                  <div
                    className="jsx-2650881538 content"
                  >
                    <div
                      className="jsx-826859815 mapSidebar"
                    />
                    <div
                      className="jsx-826859815 mapContent"
                    >
                      <div
                        className="jsx-826859815 mapRegion"
                        id="mapRegion"
                      />
                    </div>
                  </div>
                </div>
                <JSXStyle
                  css=".content.jsx-2650881538{max-width:840px;margin:auto;}.scrollWrap.jsx-2650881538{position:absolute;top:60px;left:0;right:0;bottom:0;overflow:auto;}h1.jsx-2650881538{text-align:center;}"
                  styleId="2650881538"
                />
              </div>
            </Layout>
            <JSXStyle
              css=".mapRegionWrap.jsx-826859815{height:100%;position:relative;}#mapRegion.jsx-826859815{height:100%;background:grey;}.legend.jsx-826859815{position:absolute;top:0;right:0;width:100px;border:solid black 2px;background:white;}.centerMap.jsx-826859815{position:absolute;top:10px;left:50%;width:100px;margin-left:-50px;min-height:0px;padding:2px 20px;background:white;color:black;border:solid black 2px;}.mapSidebar.jsx-826859815{position:absolute;border-right:solid black 2px;width:300px;top:0;left:0;bottom:0;overflow:auto;overflow-x:hidden;}.mapContent.jsx-826859815{position:absolute;top:0;left:300px;right:0;bottom:0;}.placeListItem.jsx-826859815{padding:10px;border-bottom:solid black 2px;height:4em;text-overflow:ellipsis;}.placeListItem.jsx-826859815:hover{background:#EEE;cursor:pointer;}"
              styleId="826859815"
            />
          </div>
        </HomePage>
      </Connect(HomePage)>
    </Provider>
  </AttachReduxHOC(Connect(HomePage))>
</ClientEntryHOC(AttachReduxHOC(Connect(HomePage)))>
`;
